drivers/ethernet/eth_stellaris.c

```diff
@@ -219,19 +219,21 @@ static void eth_stellaris_rx(const struct device *dev)
 		goto pkt_unref;
 	}
 
-	return;
+	return 0;
 
 pkt_unref:
 	net_pkt_unref(pkt);
 
 err_mem:
 	eth_stellaris_rx_error(iface);
+	return -EIO;
 }
 
 static void eth_stellaris_isr(const struct device *dev)
 {
 	struct eth_stellaris_runtime *dev_data = dev->data;
 	int isr_val = sys_read32(REG_MACRIS);
+	int np = 0;
```

Comment:
Setting value for this variable here is not needed as we are setting it already below.